### PR TITLE
feat: two-tier session status detection with configurable patterns

### LINKED ISSUE #38: Session status does not detect "Waiting for input" state
## Description

The session status indicator fails to detect or display the "Waiting for input" state, which is the most common and important status for interactive AI agent sessions.

## Expected behavior

When a session is idle and waiting for user input, the status should clearly reflect "Waiting for input" (or equivalent).

## Current behavior

The status does not recognize this state, making it difficult to tell at a glance which sessions need attention.

### PR DESCRIPTION
**Pull request by lucascaro**
## Summary

- Adds infrastructure for detecting whether agent sessions are running, idle, or waiting for input using pane title regex matching + content-diff + debounce fallback
- For Claude: uses pane title spinner detection (`^[⠁-⠿]` = running) with content-diff fallback. Waiting detection deferred — Claude Code uses `✳` for both idle and asking-a-question states
- Configurable per agent via `StatusDetection` in agent profiles (`wait_title`, `run_title`, `wait_prompt`, `idle_prompt`, `stable_ticks`)
- Adds `GetPaneTitles` batch API to Backend interface for efficient pane title reads
- Fixes data race (snapshot-copy maps before tick goroutine), memory leak (cleanup on session kill), and ANSI handling (strip before emptiness check)

## Test plan

- [x] `go build ./...` passes
- [x] `go vet ./...` passes
- [x] `go test ./...` passes — 17 new tests covering all detection paths, ANSI edge cases, migration backfill, and mock backend
- [ ] Manual: run hive with Claude sessions, confirm idle sessions show idle dot (not waiting), running sessions show green dot

Fixes #38

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/config/migrate.go [hunk 1]
```diff
@@ -11,14 +11,24 @@ func Migrate(cfg Config) Config {
...
 				profile.InstallCmd = def.InstallCmd
-				cfg.Agents[name] = profile
+				changed = true
 			}
 		}
+		if profile.Status.StableTicks == 0 {
+			if def, ok := defaults.Agents[name]; ok {
+				profile.Status = def.Status
+				changed = true
+			}
```
[the remark this answers is not in the document]
Comment (lucascaro) [reply internal/config/migrate.go]: Fixed in 93f2739 — migration now checks the entire StatusDetection struct is zero-valued instead of just StableTicks==0, so user-specified patterns are preserved.
